### PR TITLE
[Poromechanics] Linear solvers names have been updated to current standard

### PR DESCRIPTION
**Pull request by ipouplana**
Linear solvers names have been updated according to the current nomenclature of python_linear_solver_factory.

@philbucher , I still have the following warning when running my tests:

```
Linear-Solver-Factory: DEPRECATION-WARNING:
The linear-solver "super_lu" is defined in the "ExternalSolversApplication"
Please specify the "solver_type" including the name of the application:
"ExternalSolversApplication.super_lu"
Please update your settings accordingly, the current settings are deprecated!
```

Should I also change the "super_lu" name by "ExternalSolversApplication.super_lu" ?

### CODE REVIEW

File: applications/PoromechanicsApplication/tests/element_tests/undrained_soil_column_2D/ProjectParameters.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
         "nonlocal_damage":                    false,
         "characteristic_length":              0.05,
         "linear_solver_settings":             {
-            "solver_type":   "SuperLUSolver"
+            "solver_type":   "super_lu"
```
Comment (philbucher) [inline applications/PoromechanicsApplication/tests/element_tests/undrained_soil_column_2D/ProjectParameters.json]:
```suggestion
            "solver_type":   "ExternalSolversApplication.super_lu"
```
@ipouplana you are right, the application in which the linear-solver is defined should be prepended to the name of the solver (currently this is necessary for solvers from the ExternalSolversApplication and the EigenSolversApplication)

The reason for this is that then the python-linear-solver-factory can import the application that is needed and you don't have to import those apps in the MainScript
Pretty much the same as how the ProcessFactory works

File: applications/PoromechanicsApplication/custom_problemtype/Poromechanics_Application.gid/ProjectParameters.tcl [hunk 2]
```diff
@@ -98,25 +98,25 @@ proc WriteProjectParameters { basename dir problemtypedir TableDict} {
...
             puts $FileVar "            \"max_iteration\":       100,"
             puts $FileVar "            \"scaling\":             [GiD_AccessValue get gendata Scaling],"
-            puts $FileVar "            \"preconditioner_type\": \"ILU0Preconditioner\""
-        } elseif {([GiD_AccessValue get gendata Solver_Type] eq "SkylineLUFactorizationSolver") || ([GiD_AccessValue get gendata Solver_Type] eq "SuperLUSolver")} {
+            puts $FileVar "            \"preconditioner_type\": \"ilu0\""
+        } elseif {([GiD_AccessValue get gendata Solver_Type] eq "skyline_lu_factorization") || ([GiD_AccessValue get gendata Solver_Type] eq "super_lu")} {
             puts $FileVar "            \"solver_type\":   \"[GiD_AccessValue get gendata Solver_Type]\""
         } else {
-            puts $FileVar "            \"solver_type\":   \"SuperLUSolver\""
+            puts $FileVar "            \"solver_type\":   \"super_lu\""
```
Comment (philbucher) [inline applications/PoromechanicsApplication/custom_problemtype/Poromechanics_Application.gid/ProjectParameters.tcl]: in this file you should also add it

File: applications/PoromechanicsApplication/custom_problemtype/Poromechanics_Application.gid/Poromechanics_Application.prb [hunk 3]
```diff
@@ -103,9 +103,9 @@ QUESTION: Move_Mesh#CB#(true,false)
 VALUE: false
 QUESTION: Block_Builder#CB#(true,false)
 VALUE: true
-QUESTION: Solver_Type#CB#(SkylineLUFactorizationSolver,SuperLUSolver,BICGSTABSolver,AMGCL,Klu,AztecSolver,AmgclMPISolver,MultiLevelSolver)
-VALUE: AMGCL
-HELP: OpenMP solvers: SkylineLUFactorizationSolver (Direct), SuperLUSolver (Direct), BICGSTABSolver (Iter.) and AMGCL (Iter.). MPI solvers: Klu,AztecSolver,AmgclMPISolver,MultiLevelSolver.
+QUESTION: Solver_Type#CB#(skyline_lu_factorization,super_lu,bicgstab,amgcl,Klu,AztecSolver,AmgclMPISolver,MultiLevelSolver)
+VALUE: amgcl
+HELP: OpenMP solvers: skyline_lu_factorization (Direct), super_lu (Direct), bicgstab (Iter.) and amgcl (Iter.). MPI solvers: Klu,AztecSolver,AmgclMPISolver,MultiLevelSolver.
```
Comment (philbucher) [inline applications/PoromechanicsApplication/custom_problemtype/Poromechanics_Application.gid/Poromechanics_Application.prb]: the names for the trilinos-solvers are also updated, see [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/trilinos_application/custom_factories/trilinos_linear_solver_factory.cpp#L40-L78)

**Review by philbucher — COMMENTED** [review]
some minor comments otherwise 👍

File: applications/PoromechanicsApplication/tests/element_tests/undrained_soil_column_2D/ProjectParameters.json (hunk 1 above)
Comment (ipouplana) [reply applications/PoromechanicsApplication/tests/element_tests/undrained_soil_column_2D/ProjectParameters.json]: Ok, thanks. I will adapt this PR accordingly

File: applications/PoromechanicsApplication/custom_problemtype/Poromechanics_Application.gid/Poromechanics_Application.prb (hunk 3 above)
Comment (ipouplana) [reply applications/PoromechanicsApplication/custom_problemtype/Poromechanics_Application.gid/Poromechanics_Application.prb]: Ok, I will also update the MPI solvers names then

**Review by philbucher — APPROVED** [review]
Perfect !

**Comment by philbucher** [comment]
Btw sorry for the multiple changes in the recent past
Now hopefully nothing will change regarding this for a while
At least nothing is planned

**Pull request merged**